### PR TITLE
Update test deps

### PR DESCRIPTION
**Pull request by creviera**
## Description

No longer specify dependencies of dependencies in `test-requirements.in` and use the latest `semgrep`.

## Test

1. `git checkout update-test-deps`
2. `deactivate && rm -r .venv`
3. `make venv`
4. `source .venv/bin/activate`
- [ ] `make check` passes

### CODE REVIEW

**Comment by creviera** [comment]
Also notice that in this PR we are using the latest pip-tools and running with the `--allow-unsafe` option when updating test requirements via `make update-pip-requirements`. So in addition to the test plan above, test updating pip requirements by following these steps:

1. `rm requirements/test-requirements.txt`
2. `make update-pip-requirments`
3. `deactivate && rm -r .venv`
- [ ] `make venv` runs without error

**Comment by creviera** [comment]
To see why `--allow-unsafe` is needed, follow these steps:

1. checkout this PR branch
2. remove `--allow-unsafe ` and save in the Makefile
3. `rm -r requirements/test-requirements.txt`
4. `make update-pip-requirements`
5. `deactivate`
6. `rm -r .venv`
7. `make venv`

Expected
No errors

Actual
```
In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
    pip>=20.3 from https://files.pythonhosted.org/packages/ca/bf/4133a0e05eac641ec270bbcef30512b5ad307d7838adb994acd652cc30e3/pip-21.2.3-py3-none-any.whl#sha256=895df6014c2f02f9d278a8ad6e31cdfd312952b4a93c3068d0556964f4490057 (from pip-tools==6.2.0->-r requirements/test-requirements.txt (line 135))
make: *** [Makefile:4: venv] Error 1
```

**Comment by eloquence** [comment]
After some digging, @creviera and I were able to figure out why we see different results in terms of which dependencies have to be explicitly declared: I was using a locally installed version of `pip-tools` to generate the requirements, while Allie was using an (identical!) version first installed in the virtualenv. 

Even though they were both `6.2.0`, installing the version in the virtualenv first ensured that dependencies like `importlib-metadata` (which itself is required by `pip-tools` as well) are installed and preserved in pinned versions.

In other words, at least for these specific dependencies, as long as we always use a version of `pip-tools` installed in the `virtualenv`, removing them should be fine; this PR works as advertised for me. :)

File: Makefile [hunk 1]
```diff
@@ -10,7 +15,7 @@ safety: ## Runs `safety check` to check python dependencies for vulnerabilities
 
 .PHONY: update-pip-requirements
 update-pip-requirements: ## Updates all Python requirements files via pip-compile.
-	pip-compile --generate-hashes --output-file requirements/test-requirements.txt requirements/test-requirements.in
+	pip-compile --allow-unsafe --generate-hashes --output-file requirements/test-requirements.txt requirements/test-requirements.in
```
Comment (conorsch) [inline Makefile:18]: Nit: do we want `--upgrade` here? These are test reqs, so running the target should always go-get-the-latest, right?

**Review by conorsch — APPROVED** [review]
Looks great. For the test reqs, I assume we'd want the `--upgrade` flag to be present. But let's handle that as part of a general docs update, so we can make all the necessary tweaks in one fell swoop to adhere to policy.

**Pull request merged**